minion/frontend/views.py

```diff
@@ -414,6 +426,70 @@ def persona_login():
     session['role'] = user['role']
     return api_session()
 
+def ldap_login(request):
+    data = json.loads(request.data)
+    username = data['user']
+    password = data['password']
+
+    # Initialize LDAP
+    try:
+        ld = ldap.initialize(config['login_conf']['ldap_uri'])
+    except ldap.LDAPError, e:
+        return jsonify(success=False, reason="error")
+
+    auth = "uid=%s," % username + config['login_conf']['ldap_base']
+
+    # Authentication
+    try:
+        ld.simple_bind_s(auth, password)
+    except ldap.LDAPError, e:
+        if e.message['desc'] == 'Invalid credentials':
+            return jsonify(success=False, reason="invalid_cred")
+        return jsonify(success=False, reason="error")
+
+    # Retrieve mail
+    searchScope = ldap.SCOPE_SUBTREE
+    retrieveAttributes = ["mail"]
+    results = ld.search_s(auth, searchScope, attrlist=retrieveAttributes)
+
+    try:
+        mail = results[0][1]['mail'][0]
```

Comment:
This (plus other filters below) should be part of the LDAP configuration, and not buried in the function.  Off the top of my head, 'mail' is sometimes 'email' or 'userPrincipalName', and uid is sometimes 'samAccountName', etc.